### PR TITLE
Add GetCollisionBoxesEvent. Allows manipulation of the collision boxes of blocks for entities. MC1.11 edition.

### PR DESCRIPTION
**Pull request by iChun**
Slightly different from https://github.com/MinecraftForge/MinecraftForge/pull/3397/files but generally the same idea, for MC1.11

### CODE REVIEW

File: src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * This event is fired during {@link World#collidesWithAnyBlock(AxisAlignedBB)}
+ * and before returning the list in {@link World#getCollisionBoxes(Entity, AxisAlignedBB)}<br>
+ * <br>
+ * {@link #entity} contains the entity passed in the {@link World#getCollisionBoxes(Entity, AxisAlignedBB)}. <b>Can be null.</b> Calls from {@link World#collidesWithAnyBlock(AxisAlignedBB)} will be null.<br>
+ * {@link #aabb} contains the AxisAlignedBB passed in the method. <b>Do not modify this AABB</b><br>
```
Comment (liach) [inline src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java]: AABB themselves are immutable.

File: src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * {@link #collisionBoxesList} contains the list of detected collision boxes intersecting with {@link #aabb}.<br>
```
Comment (liach) [inline src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java]: You probably should mention that this list can be modified.

File: src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java (hunk 1 above)
Comment (iChun) [reply src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java]: Oh huh, I guess this became a thing when they stopped pulling from a pooled resource? Good to know, will fix.

File: src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java (hunk 2 above)
Comment (iChun) [reply src/main/java/net/minecraftforge/event/world/GetCollisionBoxesEvent.java]: That's kinda the point for this event but sure, why not.

**Pull request merged**